irohad/main/iroha-main.cpp

```diff
@@ -14,26 +14,62 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-#include <torii/command_service.hpp>
-#include <torii/query_service.hpp>
+#include <network/network_api.h>
 #include <consensus/connection/service.hpp>
-#include <ordering/connection/service.hpp>
+#include <consensus/consensus_service_stub.hpp>
+#include <main/application.hpp>
+#include <network/peer_communication_stub.hpp>
+#include <ordering/ordering_service_stub.hpp>
+#include <torii/processor/client_processor_stub.hpp>
+#include <torii/torii_stub.hpp>
+#include <validation/chain/validator_stub.hpp>
+#include <validation/stateful/stub_validator.hpp>
+#include <validation/stateless/validator_stub.hpp>
+
+#include <dao/dao.hpp>
+#include <ametsuchi/ametsuchi_stub.hpp>
+#include <dao/dao_crypto_provider_stub.hpp>
 
 #include "server_runner.hpp"
 
-int main(int argc,char* argv[]) {
-/*
-  connection::api::CommandService commandService;
-  connection::api::QueryService queryService;
-  consensus::connection::SumeragiService sumeragiService;
-  ordering::connection::OrderingService orderingService;
-
-  ServerRunner serverRunner("0.0.0.0", 50051, {
-      &commandService,
-      &queryService,
-      &sumeragiService,
-      &orderingService
-  });
-*/
+int main(int argc, char *argv[]) {
+  /*
+    connection::api::CommandService commandService;
+    connection::api::QueryService queryService;
+    consensus::connection::SumeragiService sumeragiService;
+    ordering::connection::OrderingService orderingService;
+
+    ServerRunner serverRunner("0.0.0.0", 50051, {
+        &commandService,
+        &queryService,
+        &sumeragiService,
+        &orderingService
+    });
+  */
+//  iroha::Irohad irohad;
+  iroha::ametsuchi::AmetsuchiStub ametsuchi;
+  iroha::dao::DaoCryptoProviderStub crypto_provider;
+  iroha::validation::StatelessValidatorStub stateless_validator;
+  iroha::validation::StatefulValidatorStub stateful_validator;
+  iroha::validation::ChainValidatorStub chain_validator;
+  iroha::ordering::OrderingServiceStub ordering_service;
+  iroha::consensus::ConsensusServiceStub consensus_service;
+  iroha::network::PeerCommunicationServiceStub peer_communication_service(
+      ametsuchi, stateful_validator, chain_validator, ordering_service,
+      consensus_service, crypto_provider);
+  iroha::torii::ClientProcessorStub client_processor(
+      stateless_validator, peer_communication_service, crypto_provider);
+  iroha::torii::ToriiStub torii(client_processor);
+  // shows required order of execution, since callbacks are called synchronously
+  peer_communication_service.subscribe_on_proposal();
```

Comment:
Correct, but in Irohad was broken initialization. And all services temporarily moved in main. But good news is there is no singleton in iroha! All dependencies are clear and passed only via constructor.